api/src/main/java/org/pmiops/workbench/genomics/GenomicExtractionService.java

```diff
@@ -170,9 +171,10 @@ public List<GenomicExtractionJob> getGenomicExtractionJobs(
                   }
 
                   if (isTerminal(status)) {
-                    dbSubmission.setCompletionTime(
-                        CommonMappers.timestamp(
-                            firecloudSubmission.getWorkflows().get(0).getStatusLastChangedDate()));
+                    OffsetDateTime completionTime =
```

Comment:
Broke retrieving value from response, converting, and storing into three different steps for readability and ease of debugging.